synapseclient/api/entity_services.py

```diff
@@ -0,0 +1,173 @@
+            the `.login()` method.
+
+    Returns:
+        The requested entity bundle matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Entity.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}",
+    )
+
+
+@alru_cache(ttl=60)
+async def get_upload_destination(
+    entity_id: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Union[str, int]]:
+    """
+    <https://rest-docs.synapse.org/rest/GET/entity/id/uploadDestination.html>
+
+    Arguments:
+        entity_id: The ID of the entity.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        The upload destination.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/UploadDestination.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}/uploadDestination",
+        endpoint=client.fileHandleEndpoint,
+    )
+
+
+async def get_upload_destination_location(
+    entity_id: str, location: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Union[str, int]]:
+    """
+    <https://rest-docs.synapse.org/rest/GET/entity/id/uploadDestination/storageLocationId.html>
+
+    Arguments:
+        entity_id: The ID of the entity.
+        location: A storage location ID of the upload destination.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        The upload destination.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/UploadDestination.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(
+        uri=f"/entity/{entity_id}/uploadDestination/{location}",
+        endpoint=client.fileHandleEndpoint,
+    )
+
+
+async def create_access_requirements_if_none(
+    entity_id: str, synapse_client: Optional["Synapse"] = None
+) -> None:
+    """
+    Checks to see if the given entity has access requirements. If not, then one is added
+
+    Arguments:
+        entity_id: The ID of the entity.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    existing_restrictions = await client.rest_get_async(
+        f"/entity/{entity_id}/accessRequirement?offset=0&limit=1"
+    )
+    if len(existing_restrictions["results"]) <= 0:
+        access_requirements = await client.rest_post_async(
```

Comment:
It would be nice if we were given the ability to use these endpoints but not get all the emails the ACL team gets.

Reply:
Can you clarify what you mean? Are you suggesting a change to this [REST API](https://rest-docs.synapse.org/rest/POST/entity/id/lockAccessRequirement.html)?

Reply:
So currently only members of a Synapse ACL Team can execute these endpoints.  At least to my knowledge, our user accounts don't have permission to execute these.

Reply:
Interesting!  I do know that there are specific endpoints that only ACT team members can execute - maybe that's not one of them.  Sorry about that!